### PR TITLE
Avoid runtime static parameters inference during dynamic dispatch of `SVector{k}`

### LINKED ISSUE #1148: Potentially address performance regression with Julia 1.9
There's a performance issue with Julia 1.9-rc. When using the constructors with a length determined at runtime, Julia falls back on dynamic dispatch.

```
using BenchmarkTools
using StaticArrays
const xx = [1; 2]
@noinline function e(x)
    k = length(x)
    SVector{k}(x)
end

@btime e(xx) 
# 1.9-rc1: 27.898 μs (31 allocations: 2.66 KiB)
# 1.8.5: 762.239 ns (10 allocations: 544 bytes)
```

According to https://github.com/JuliaLang/julia/issues/49145, there may be changes in StaticArrays that would work around this issue.

### PR DESCRIPTION
**Pull request by BioTurboNick**
Fixes #1148 by implementing @N5N3 's suggestion.

### CODE REVIEW

**Comment by BioTurboNick** [comment]
I'm gathering the nightly failures are unrelated to this PR, since they also show up in a more recent PR.

**Review by mateuszbaran — APPROVED** [review]
LGTM :+1: . Could you just bump patch version?

File: Project.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name = "StaticArrays"
 uuid = "90137ffa-7385-5640-81b9-e52037218182"
-version = "1.5.21"
+version = "1.5.22"
```
Comment (mateuszbaran) [inline Project.toml:3]: 1.5.22 is already released so it needs to be bumped to 1.5.23 now :slightly_smiling_face: . I think you need to either merge master into your branch or rebase this branch to get a clean merge.

Comment (BioTurboNick) [reply Project.toml:3]: *shakes fist* haha will do

Comment (BioTurboNick) [reply Project.toml]: Done!

Comment (mateuszbaran) [reply Project.toml]: Thanks!

**Pull request merged**